### DISCUSSION TITLE
Error when building docker image with extensions

### DISCUSSION THREAD

**Discussion by nsaliu** [serversideup/docker-php#383] [Q&A] [answered]
Hello everyone,
I was trying to build the following docker image:

```
FROM serversideup/php:8.3.6-fpm-nginx as base
USER root
RUN install-php-extensions intl bcmath soap
USER www-data
```

but I'm receiving this error 💥 👇 

```
 => ERROR [base 2/2] RUN install-php-extensions intl bcmath soap                                                                                                             8.1s
------
 > [base 2/2] RUN install-php-extensions intl bcmath soap:
0.071 install-php-extensions v.2.2.6
0.071 #StandWithUkraine
1.024 Updating channel "pecl.php.net"
1.287 Channel "pecl.php.net" is up to date
1.372 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
1.411 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
1.428 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
1.444 Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
1.620 Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [13.7 kB]
1.620 Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [157 kB]
7.626 Get:7 http://nginx.org/packages/debian bookworm InRelease [2866 B]
7.673 Err:7 http://nginx.org/packages/debian bookworm InRelease
7.673   The following signatures were invalid: EXPKEYSIG ABF5BD827BD9BF62 nginx signing key <signing-key@nginx.com>
7.680 Reading package lists...
7.960 W: GPG error: http://nginx.org/packages/debian bookworm InRelease: The following signatures were invalid: EXPKEYSIG ABF5BD827BD9BF62 nginx signing key <signing-key@nginx.com>
7.960 E: The repository 'http://nginx.org/packages/debian bookworm InRelease' is not signed.
```

Any ideas how to solve it?
Thanks in advance!

**Comment by jaydrogers** [comment]
Make sure you're doing a `docker pull serversideup/php:8.3.6-fpm-nginx`.

There was a bug last week that was fixed by re-running the build: https://github.com/serversideup/docker-php/issues/380

**Reply by jaydrogers** [reply, marked as answer]
### Update: It's a deeper issue
The image `serversideup/php:8.3.6-fpm-nginx` was built with a version older than `v3.2.0`.

### Workaround
Use `serversideup/php:8.3.8-fpm-nginx` or update the NGINX key manually in your Dockerfile:

https://github.com/serversideup/docker-php/blob/11e8235ba2de6a6c858737c91c61a7520f1e548b/src/variations/fpm-nginx/Dockerfile#L25-L49

**Reply by nsaliu** [reply]
Thanks for the reply!

> Use serversideup/php:8.3.8-fpm-nginx

It works, so the image is built.

BUT, there is a new issue that prevents me running the containers:
- I'm using `docker swarm` to run the containers
- when starting the stack in docker swarm, the container is created and it goes in the `healthy` status, but right after the container is turned off without reaching the `running` status.
- it was not happening with the `php8.3.6-fpm-nginx` image
- the logs are OK, so no errors at all, just the container turned OFF without any advice or warning

@jaydrogers is there a way to release a fix from your side for the `8.3.6-fpm-nginx`?

**Reply by nsaliu** [reply]
Update: after more than 10 restarts the containers are running smoothly and the status is always healthy (the old `8.3.6` are in the `running` status).

That's strange, using the `8.3.6-fpm-nginx` before this issue I had zero restarts and the container was running in the first shot.
